include/aikido/planner/dart/ConfigurationToEndEffectorOffset.hpp

```diff
@@ -12,31 +13,94 @@ namespace planner {
+  /// on construction.
+  ///
+  /// \param[in] stateSpace State space.
+  /// \param[in] startState Start state to plan from.
+  /// \param[in] endEffectorBodyNode BodyNode to be planned to move to a desired
+  /// offest while maintaining the current orientation.
+  /// \param[in] direction Unit vector that represents the direction of motion
+  /// [unit vector in the world frame].
+  /// \param[in] signedDistance Signed distance to move, in meters.
+  /// \param[in] constraint Trajectory-wide constraint that must be satisfied.
+  /// \throw If the size of \c direction is zero.
+  ConfigurationToEndEffectorOffset(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr stateSpace,
       const statespace::dart::MetaSkeletonStateSpace::State* startState,
+      ::dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
       const Eigen::Vector3d& direction,
       double signedDistance,
       constraint::ConstTestablePtr constraint);
 
+  /// Constructor. Note that this constructor takes the start state from the
+  /// current state of the passed MetaSkeleton, and takes the direction from the
+  /// end-effector BodyNode's current direction.
+  ///
+  /// \param[in] stateSpace State space.
+  /// param[in] metaSkeleton MetaSkeleton that getStartState will return the
+  /// current state of when called.
+  /// \param[in] endEffectorBodyNode BodyNode to be planned to move to a desired
+  /// offest while maintaining the current orientation. Current direction is
+  /// returned when getDirection is called.
+  /// \param[in] signedDistance Signed distance to move, in meters.
+  /// \param[in] constraint Trajectory-wide constraint that must be satisfied.
+  ConfigurationToEndEffectorOffset(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr stateSpace,
+      ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton,
+      ::dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
+      double signedDistance,
+      constraint::ConstTestablePtr constraint);
+
+  /// Constructor. Note that this constructor sets the start state on
+  /// construction, and takes the direction from the end-effector BodyNode's
+  /// current direction.
+  ///
+  /// \param[in] stateSpace State space.
+  /// \param[in] startState Start state to plan from.
+  /// \param[in] endEffectorBodyNode BodyNode to be planned to move to a desired
+  /// offest while maintaining the current orientation. Current direction is
+  /// returned when getDirection is called.
+  /// \param[in] signedDistance Signed distance to move, in meters.
+  /// \param[in] constraint Trajectory-wide constraint that must be satisfied.
+  ConfigurationToEndEffectorOffset(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr stateSpace,
+      const statespace::dart::MetaSkeletonStateSpace::State* startState,
+      ::dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
+      double signedDistance,
+      constraint::ConstTestablePtr constraint);
```

Comment:
Nit: Let's pass `nullptr` to `constraint` as the default value for the same reason.